<launch>
  <!-- only SE2 is supported -->
  <arg name="sampling_space" value="SE2" />

  <node pkg="differentiable_rmap" type="NodeRmapPlanningFootstep" name="rmap_planning_footstep"
        output="screen">
    <rosparam subst_value="true">
      sampling_space: $(arg sampling_space)
      config_path: $(find differentiable_rmap)/config/RmapPlanningFootstep.yaml
      svm_path: $(find differentiable_rmap)/data/rmap_svm_model_$(arg sampling_space)_footstep.libsvm
      bag_path: $(find differentiable_rmap)/data/rmap_grid_set_$(arg sampling_space)_footstep.bag
    </rosparam>
  </node>

  <node pkg="optmotiongen_core" type="InteractiveMarkerManager.py" name="rviz_server"
        clear_params="true">
    <rosparam>
      interactive_markers:
        - name: target
          initial_pose:
            pos: [0, 0, 0.1]
          axis_controls: [tx, ty, rz]
          markers:
            - type: 0 # ARROW
              scale: [0.15, 0.05, 0.05]
              color: [1.0, 1.0, 0.0, 1.0]
        - name: start
          initial_pose:
            pos: [0, 0, 0.1]
          axis_controls: []
          markers:
            - type: 0 # ARROW
              scale: [0.15, 0.05, 0.05]
              color: [1.0, 1.0, 0.0, 1.0]
    </rosparam>
  </node>

  <node pkg="rviz" type="rviz" name="$(anon rviz)"
        args="-d $(find differentiable_rmap)/rviz/RmapPlanningFootstep.rviz"
        output="log" clear_params="true" />
</launch>
